### PR TITLE
Fix tiff io plugin

### PR DESCRIPTION
**Pull request by ericpre**
### Description of the change
Fix issue related to last release of the tifffile library, as mentioned by @tjof2 on https://gitter.im/hyperspy/hyperspy?at=5e5005293ca8a67fb80cf5c5.
See for example:
https://github.com/hyperspy/hyperspy/pull/2321, https://travis-ci.org/hyperspy/hyperspy/builds/653503166
and
https://github.com/hyperspy/hyperspy/pull/2322, https://travis-ci.org/hyperspy/hyperspy/builds/653999456

This issue is only in the development branches, since currently the release version does not depends on the tifffile library.

### Progress of the PR
- [x] Fix 
- [x] Fix deprecation functions,
- [x] ready for review.

### CODE REVIEW

**Comment by thomasaarholt** [comment]
Didn't see this before I created #2325.

**Comment by dnjohnstone** [comment]
This looks good to me - please can someone merge it?

**Review by sem-geologist — APPROVED** [review]
Looks good for me

File: hyperspy/io_plugins/tiff.py [hunk 1]
```diff
@@ -159,7 +161,11 @@ def file_reader(filename, record_by='image', force_read_resolution=False,
             dtype = np.dtype({'names': names[:lastshape],
                               'formats': [dtype] * lastshape})
             shape = shape[:-1]
-        op = {tag.name: tag.value for tag in tiff.pages[0].tags}
+        if LooseVersion(tifffile.__version__) >= LooseVersion("2020.2.16"):
```
Comment (sem-geologist) [inline hyperspy/io_plugins/tiff.py:164]: Oh, this. Are you sure that is version the behaviour changes of tiffiles?

Comment (thomasaarholt) [reply hyperspy/io_plugins/tiff.py:164]: Based on the [GitHub releases page](https://github.com/cgohlke/tifffile/releases) and some quick testing, this check looks correct.

**Comment by thomasaarholt** [comment]
LGTM2, would appreciate merged so that failures of other PRs becomes clearer.

**Comment by tjof2** [comment]
Agreed

**Comment by ericpre** [comment]
Thanks for the review everyone! :)

**Pull request merged**